homeassistant/components/homekit/thermostats.py

```diff
@@ -0,0 +1,137 @@
+
+        self.service_thermostat = add_preload_service(self, SERV_THERMOSTAT)
+        self.char_current_heat_cool = self.service_thermostat.\
+            get_characteristic(CHAR_CURRENT_HEATING_COOLING)
+        self.char_target_heat_cool = self.service_thermostat.\
+            get_characteristic(CHAR_TARGET_HEATING_COOLING)
+        self.char_target_heat_cool.setter_callback = self.set_heat_cool
+
+        self.char_current_temp = self.service_thermostat.\
+            get_characteristic(CHAR_CURRENT_TEMPERATURE)
+        self.char_target_temp = self.service_thermostat.\
+            get_characteristic(CHAR_TARGET_TEMPERATURE)
+        self.char_target_temp.setter_callback = self.set_temperature_debounced
+
+        self.char_display_units = self.service_thermostat.\
+            get_characteristic(CHAR_TEMP_DISPLAY_UNITS)
+
+    def run(self):
+        """Method called be object after driver is started."""
+        state = self._hass.states.get(self._entity_id)
+        self.update_thermostat(new_state=state)
+
+        async_track_state_change(self._hass, self._entity_id,
+                                 self.update_thermostat)
+
+    def set_heat_cool(self, value):
+        """Move operation mode to value if call came from HomeKit."""
+        if value != self.current_heat_cool and value in HC_HOMEKIT_TO_HASS:
+            _LOGGER.debug("%s: Set heat-cool to %d", self._entity_id, value)
+            self.homekit_target_heat_cool = value
+            hass_value = HC_HOMEKIT_TO_HASS[value]
+
+            self._hass.services.call('climate', 'set_operation_mode',
+                                     {'entity_id': self._entity_id,
+                                      'operation_mode': hass_value})
+
+    def set_temperature_debounced(self, value):
+        _LOGGER.debug("%s: Call to set temperature to %.2f",
+                      self._entity_id, value)
+
+        def perform_call():
+            self.set_target_temperature(value)
+        # Cancel timer if one was running
+        if self._call_timer is not None:
+            try:
+                self._call_timer.cancel()
+            except AttributeError:
+                pass
+        self._call_timer = Timer(1, perform_call)
+        self._call_timer.start()
+
+    def set_target_temperature(self, value):
+        """Set target temperature to value if call came from HomeKit."""
+        if value != self.target_temperature:
+            _LOGGER.debug("%s: Set target temperature to %.2f",
+                          self._entity_id, value)
+            self.homekit_target_temperature = value
+            self._hass.services.call('climate', 'set_temperature',
+                                     {'entity_id': self._entity_id,
+                                      'temperature':
+                                          self.homekit_target_temperature})
+
+    def update_thermostat(self, entity_id=None,
+                          old_state=None, new_state=None):
```

Comment:
I haven't look at this method in detail, but it might be a good idea to use flag vars as well, like I suggested above.